pkg/rancher-desktop/config/settings.ts

```diff
@@ -96,7 +96,12 @@ export const defaultSettings = {
      */
     hostResolver: true,
   },
-  WSL:        { integrations: {} as Record<string, boolean> },
+  WSL: {
+    integrations: {} as Record<string, boolean>,
+    proxy:        {
```

Comment:
We should probably confirm with @jandubois on where the proxy config schema should go?

Reply:
They should go under `virtualMachine` because they are applicable on other platforms too.

Having a separate `WSL` object at the top level is kind of weird, as in many ways it is just the Windows-specific term for `virtualMachine`.

So in Preferences we are going to have a "Virtual Machine | Networking" tab on macOS and Linux, but on Windows it will be "WSL | Networking". But the setting has to live in the same spot in the schema on all platforms. So only settings that are only relevant to Windows can go into the `WSL` object.

Reply:
@jandubois Make sense. Do you want me to rename the `WSLProxy.ts` to something else already ?

Reply:
I just realized that we have already work in progress adding a "WSL | Network" tab (part of #4181), so we'll need to coordinate these changes.

I think calling it `WSLProxy` is fine, as the implementation is WSL-specific. I just wanted the `proxy` **setting** to be part of the `virtualMachine` object, so we don't need to change that later.